rewrite-core/src/test/java/org/openrewrite/remote/RemoteArchiveTest.java

```diff
@@ -52,6 +54,27 @@ void gradleWrapper(String version) throws Exception {
         assertThat(actual).isGreaterThan(50_000);
     }
 
+    @Test
+    void gradleWrapperDownloadFails() throws Exception {
+        URL distributionUrl = requireNonNull(RemoteArchiveTest.class.getClassLoader().getResource("gradle-7.4.2-bin.zip"));
+        ExecutionContext ctx = new InMemoryExecutionContext();
+        HttpSenderExecutionContextView.view(ctx)
+          .setLargeFileHttpSender(new MockHttpSender(408));
+
+        RemoteArchive remoteArchive = Remote
+          .builder(
+            Paths.get("gradle/wrapper/gradle-wrapper.jar"),
+            distributionUrl.toURI()
+          )
+          .build("gradle-[^\\/]+\\/(?:.*\\/)+gradle-wrapper-(?!shared).*\\.jar");
+
+
+        assertThat(
+          assertThrows(IllegalStateException.class, () -> getInputStreamSize(remoteArchive.getInputStream(ctx)))
+            .getMessage()
+        ).contains("Failed to download");
```

Comment:
This nested JUnit inside AssertJ reads a bit odd to me; can't we use AssertJ's `assertThatThrownBy` to directly assert the message?

Reply:
`assertThatThrownBy` is much better fit I will update